ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzerFactory.java

```diff
@@ -94,17 +96,11 @@ private static BaseSemanticAnalyzer getInternal(QueryState queryState, ASTNode t
         return new ColumnStatsSemanticAnalyzer(queryState);
 
       case HiveParser.TOK_UPDATE_TABLE:
-        if (HiveConf.getBoolVar(queryState.getConf(), HiveConf.ConfVars.SPLIT_UPDATE)) {
-          return new SplitUpdateSemanticAnalyzer(queryState);
-        }
+        return new UpdateSemanticAnalyzer(queryState, new UpdateRewriterFactory(queryState.getConf()));
       case HiveParser.TOK_DELETE_FROM:
-        return new UpdateDeleteSemanticAnalyzer(queryState);
+        return new DeleteSemanticAnalyzer(queryState, new DeleteRewriterFactory(queryState.getConf()));
 
       case HiveParser.TOK_MERGE:
-        if (HiveConf.getBoolVar(queryState.getConf(), HiveConf.ConfVars.SPLIT_UPDATE) ||
-                HiveConf.getBoolVar(queryState.getConf(), HiveConf.ConfVars.MERGE_SPLIT_UPDATE)) {
-          return new SplitMergeSemanticAnalyzer(queryState);
-        }
         return new MergeSemanticAnalyzer(queryState);
```

Comment:
dup of pev comment. why don't we follow the same approach as for Update/Delete -> pass or init in constructor?

Reply:
Added MergeRewriter